### ISSUE TITLE
makeMixinParameters throws TypeError: Cannot read property '1' of null

### ISSUE THREAD

**Issue by Ciantic** [mrmlnc/less-symbols-parser#7]
Apologies for bothering you, I know that this could as well be a abandonware since you are not using LESS anymore.

I get *TypeError: Cannot read property '1' of null* in VSCode console, they emit from here:

https://github.com/mrmlnc/less-symbols-parser/blob/c5fbb881b5a4d182acff8f031a3f6d2f14944de4/src/index.ts#L41

Reason is that line of code assumes the regex match succeeded, it should have a check that if it's null do something else.

**Comment by mrmlnc** [comment]
Hello, @Ciantic,

Can you give me an example of code that causes this error?

**Comment by Ciantic** [comment]
Did some digging, I think it's this, the `text` parameter contains `...` when the error occurs.
```
.filter(@a...) {
    -webkit-filter: @a;
    -moz-filter: @a;
    -ms-filter: @a;
    filter: @a;
}
```

However I can just throw that away I use post-css.

**Comment by Ciantic** [comment]
Or this
```

.linear-gradient-multi(...) {
    background-image: -webkit-linear-gradient(@arguments);
    background-image: -moz-linear-gradient(@arguments);
    background-image: -ms-linear-gradient(@arguments);
    background-image: -o-linear-gradient(@arguments);
    background-image: linear-gradient(@arguments);
}
```

**Comment by tianzhich** [comment]
Yes, we need to support [rest arguments](https://lesscss.org/features/#mixins-feature-advanced-arguments-and-the-rest-variable) pattern. And also, the parameter of mixins is not the same as a normal variable, it could just be an arbitrary word for [pattern matching](https://lesscss.org/features/#mixins-feature-pattern-matching).

So these two edge cases would break the params splitting and the token matching in `makeMixinParameters`:

```ts
function makeMixinParameters(text: string, offset: number): IVariable[] {
	const variables: IVariable[] = [];
	if (text === '()') {
		return variables;
	}

	// Remove parenthesis
	text = text.slice(1, text.length - 1);
	const params = text.split(/([,;]\s*)(?=@)/);

	// Skip `(`
	offset += 1;

	for (let i = 0; i < params.length; i = i + 2) {
		const token = params[i];
		const stat = token.match(/([\n\t\r\s]*)(@[\w-]+)(?:\s*:\s*(.*))?/);
	}
// ...
}
```

I've tried to modify it like this:

![image](https://user-images.githubusercontent.com/33218310/130063881-5e9300fc-7a21-4111-b26b-92e11190c0e5.png)

But there is a case would failed:

```ts
.foo(@color: rgba(0,0,0,0), bar)
```

I couldn't find a proper way to split the params if `(?=@)` this restriction is unnecessary.

Looking forward to your help @mrmlnc if you have any free time.
